addons/l10n_in/models/account_invoice.py

```diff
@@ -91,20 +91,17 @@ class AccountMove(models.Model):
     # COMPUTE METHODS
     # -------------------------------------------------------------------------
 
-    @api.depends('partner_id', 'partner_id.l10n_in_gst_treatment', 'state')
+    @api.depends('partner_id.vat', 'partner_id.country_id', 'partner_id.l10n_in_gst_treatment', 'state')
     def _compute_l10n_in_gst_treatment(self):
-        indian_invoice = self.filtered(lambda m: m.country_code == 'IN')
-        for record in indian_invoice:
-            if record.state == 'draft':
-                gst_treatment = record.partner_id.l10n_in_gst_treatment
-                if not gst_treatment:
-                    gst_treatment = 'unregistered'
-                    if record.partner_id.country_id.code == 'IN' and record.partner_id.vat:
-                        gst_treatment = 'regular'
-                    elif record.partner_id.country_id and record.partner_id.country_id.code != 'IN':
-                        gst_treatment = 'overseas'
-                record.l10n_in_gst_treatment = gst_treatment
-        (self - indian_invoice).l10n_in_gst_treatment = False
+        for invoice in self.filtered(lambda m: m.country_code == 'IN' and m.state == 'draft'):
+            partner = invoice.partner_id
+            invoice.l10n_in_gst_treatment = (
+                partner.l10n_in_gst_treatment
+                or (
+                    'overseas' if partner.country_id and partner.country_id.code != 'IN'
+                    else partner.check_vat_in(partner.vat) and 'regular' or 'consumer'
```

Comment:
With the changes in base_vat still to be pushed, it should become the case again.

Reply:
@jco-odoo  There is no difference between 'unregistered' and 'consumer'; maybe we plan to merge it

Reply:
Ok, for another PR then.